### ISSUE TITLE
Ability to set constraints on UILayoutGuide.

### ISSUE THREAD

**Issue by marcotolman** [SnapKit/SnapKit#327]
### New Issue Checklist
- [x] I have looked at the [Documentation](http://snapkit.io/docs)
- [x] I have read the [F.A.Q.](http://snapkit.io/faq)
- [x] I have filled out this issue template.
### Issue Info

| Info | Value |
| --- | --- |
| Platform | iOS |
| Platform Version | 10.0 |
| SnapKit Version | 0.40.0 |
| Integration Method | cocoapods |
### Issue Description

Maybe this is already possible and I just don't understand how to do it, but is it possible to set constraints on a UILayoutGuide using Snapkit?

so something like this:

```
let guide = UILayoutGuide()
someView.addLayoutGuide(guide)

guide.snp.makeConstraints { make in
    make.width.equalToSuperview()
    make.height.equalTo(100)
    make.top.equalToSuperview()
}
```

If this is not possible, will this functionality be added later?

Thanks!

**Comment by robertjpayne** [comment]
@marcotolman this should already be supported, you should be able to `make.top.equalTo(guide.snp.bottom)`.

**Comment by marcotolman** [comment]
@robertjpayne Yes I know that's possible! And it's great :)
According to Apple's documentation however, it's also possible to set constraints on a UILayoutGuide just like you can for UIViews. For that I would like to use Snapkit (like in the example code in my first post).
Unless I'm mistaken that's not possible at the moment. So I can constrain my views to the layout guides using Snapkit, I just can't set constraints on layout guides using Snapkit. I hope this explains my question better.

**Comment by robertjpayne** [comment]
@marcotolman I see you want a constraint maker API for a layout guide? e.g. `guide.snp.makeConstraiints`?

**Comment by marcotolman** [comment]
@robertjpayne Yes!

**Comment by robertjpayne** [comment]
@marcotolman ok, it shouldn't be too hard I'll try and get around to it. Or if you'd like to take a stab and submitting the feature as a PR you'll be after adding it to the [ConstraintLayoutGuideDSL.swift](https://github.com/SnapKit/SnapKit/blob/develop/Source/ConstraintLayoutGuideDSL.swift) file. You can use the [ConstraintViewDSL.swift](https://github.com/SnapKit/SnapKit/blob/develop/Source/ConstraintViewDSL.swift) as reference.

**Comment by robertjpayne** [comment]
Actually this may be harder than what I just let on to above ^^.  A lot of the internals are targeted around a view being the leading part of any constraint.

**Comment by marcotolman** [comment]
@robertjpayne OK. Thanks for looking at it. I'll see if I can find the time to play around with it. 
Thanks again!

**Comment by robertjpayne** [comment]
@marcotolman lets leave this open, it's good enhancement on the list 👍

**Comment by robertjpayne** [comment]
@marcotolman this is in develop now, I'd love if you could give it a whirl.

**Comment by robertjpayne** [comment]
@marcotolman you by chance get a chance to give this a go and see if it works for you?

**Comment by marcotolman** [comment]
@robertjpayne Just now saw this! Will give it a try!

**Comment by marcotolman** [comment]
@robertjpayne I gave it a quick whirl and it works great! Thanks for this, no more dummy views to layout a view ;)

**Comment by robertjpayne** [comment]
🎉 this is released in 3.1.2
